scrapy/spiderloader.py

```diff
@@ -23,6 +23,10 @@ def __init__(self, settings):
 
     def _load_spiders(self, module):
         for spcls in iter_spider_classes(module):
+            if spcls.name in self._spiders.keys():
+                import warnings
+                warnings.warn("There are several spiders with the same name (" + spcls.name +
+                              "), this can cause unexpected behavior", UserWarning)
```

Comment:
Updated.
This is how it looks like in the console now:

```
$ scrapy crawl example
/home/paul/src/scrapy/scrapy/spiderloader.py:37: UserWarning: There are several spiders with the same name 'example':
        ExampleSpider (in dupespider.spiders.example)
        ExampleSpider (in dupespider.spiders.example2)
    This can cause unexpected behavior.
  warnings.warn(msg, UserWarning)
2017-03-07 14:42:34 [scrapy.utils.log] INFO: Scrapy 1.3.2 started (bot: dupespider)
```